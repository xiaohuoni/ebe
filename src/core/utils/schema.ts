import { Component } from '@lingxiteam/types';
import * as _ from 'lodash';
import { IPublicTypeNodeDataType } from '../types';
import { CodeGeneratorError } from '../types/error';
import {
  isDOMText,
  isJSExpression,
  isJSFunction,
  isJSSlot,
  isNodeSchema,
} from '../utils/deprecated';
import { isJSExpressionFn } from './common';
export function isContainerSchema(x: any): x is any {
  return (
    typeof x === 'object' &&
    x &&
    typeof x.type === 'string' &&
    typeof x.fileName === 'string'
  );
}

export function isNpmInfo(x: any): x is any {
  return typeof x === 'object' && x && typeof x.package === 'string';
}

const noop = () => undefined;

const handleChildrenDefaultOptions = {
  rerun: false,
};

const DEFAULT_MAX_DEPTH = 100000;

/**
 * 遍历并处理所有的子节点
 * @param components
 * @param handlers
 * @param options
 * @returns
 */
export function handleSubNodes<T>(
  components: IPublicTypeNodeDataType,
  handlers: {
    string?: (i: string) => T;
    expression?: (i: any) => T;
    node?: (i: IPublicTypeNodeDataType) => T;
  },
  options?: {
    rerun?: boolean;
    maxDepth?: number; // 防止出现死循环无穷递归
  },
): T[] {
  const opt = {
    ...handleChildrenDefaultOptions,
    ...(options || {}),
  };
  const maxDepth = opt.maxDepth ?? DEFAULT_MAX_DEPTH;
  if (maxDepth <= 0) {
    throw new Error('handleSubNodes maxDepth reached');
  }

  if (Array.isArray(components)) {
    const list: any[] = components as any;
    return list
      .map((child) =>
        handleSubNodes(child, handlers, { ...opt, maxDepth: maxDepth - 1 }),
      )
      .reduce((p, c) => p.concat(c), []);
  }

  let result: T | undefined;
  const childrenRes: T[] = [];
  if (components === null || components === undefined) {
    return [];
  } else if (isDOMText(components)) {
    const handler = handlers.string || noop;
    result = handler(components as any);
  } else if (isJSExpression(components)) {
    const handler = handlers.expression || noop;
    result = handler(components);
  } else if (isJSSlot(components)) {
    // @ts-ignore
    return handleSubNodes(components.value as any, handlers, {
      ...opt,
      maxDepth: maxDepth - 1,
    });
  } else if (isNodeSchema(components)) {
    const handler = handlers.node || noop;
    const child = components as any;
    result = handler(child);

    if (child.components) {
      const childRes = handleSubNodes(child.components, handlers, opt);
      childrenRes.push(...childRes);
    }

    if (child.props) {
      if (Array.isArray(child.props)) {
        child.props.forEach(({ value }: any) => {
          const childRes = handleCompositeValueInProps(value);
          childrenRes.push(...childRes);
        });
      } else {
        Object.values(child.props).forEach((value) => {
          const childRes = handleCompositeValueInProps(value as any);
          childrenRes.push(...childRes);
        });
      }
    }
  } else {
    throw new CodeGeneratorError(
      'handleSubNodes got invalid NodeData',
      components,
    );
  }

  if (result !== undefined) {
    childrenRes.unshift(result);
  }

  return childrenRes;

  function handleCompositeValueInProps(value: any): T[] {
    if (isJSSlot(value)) {
      return handleSubNodes(value.value, handlers, {
        ...opt,
        maxDepth: maxDepth - 1,
      });
    }

    // CompositeArray
    if (Array.isArray(value)) {
      return _.flatMap(value, (v) => handleCompositeValueInProps(v));
    }

    // IPublicTypeCompositeObject
    if (
      !isJSExpression(value) &&
      !isJSExpressionFn(value) &&
      !isJSFunction(value) &&
      typeof value === 'object' &&
      value !== null
    ) {
      return _.flatMap(Object.values(value), (v) =>
        handleCompositeValueInProps(v),
      );
    }

    return [];
  }
}

export function isValidContainerType(schema: Component) {
  return ['Page', 'Component', 'Block'].includes(schema.type);
}

export const enum ContainerType {
  Page = 'Page',
  Component = 'Component',
  Block = 'Block',
}
