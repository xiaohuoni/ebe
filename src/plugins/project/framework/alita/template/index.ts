import { IProjectTemplate, ResultDir } from '../../../../../core';
import { generateStaticFiles } from './static-files';

const alitaTemplate: IProjectTemplate = {
  slots: {
    pages: {
      path: ['src', 'pages'],
    },
    components: {
      path: ['src', 'components'],
    },
    packageJSON: {
      path: [],
      fileName: 'package',
    },
    appConfig: {
      path: ['config'],
      fileName: 'config',
    },
    pageview: {
      path: ['src', 'components', 'Pageview'],
    },
    app: {
      path: ['src'],
      fileName: 'app',
    },
  },

  generateTemplate(parseResult, postProcessors): ResultDir {
    return generateStaticFiles(parseResult, postProcessors);
  },
};

export default alitaTemplate;
