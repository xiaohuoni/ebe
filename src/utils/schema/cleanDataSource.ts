function cleanTree(tree: any, fields: string[]): any {
  let fieldSet = new Set(fields); // 使用set结构可以提高查询速度

  if (Array.isArray(tree)) {
    return tree.map((item: any) => cleanTree(item, fields));
  } else if (typeof tree === 'object' && tree !== null) {
    return Object.entries(tree).reduce((newTree: any, [key, value]) => {
      if (!fieldSet.has(key)) {
        newTree[key] = cleanTree(value, fields);
      }
      return newTree;
    }, {});
  } else {
    return tree;
  }
}

export const cleanDataSource = (dataSource: any[]) => {
  return dataSource.map((item) => {
    switch (item?.source) {
      // {
      //     "id": 166780606235440800,
      //     "name": "$_hxLabelDataList",
      //     "description": "画像标签数据集合",
      //     "source": "custom",
      //     "type": "objectArray",
      //     "filterParams": [],
      //     "outParams": [],
      //     "config": { "hooks": [] }
      //   },
      // item: {name: string, outParams?: never[] | undefined; type?: 'object' | 'objectArray'; rootOutParams: any; }
      case 'custom': {
        return cleanTree(item, ['id', 'description', 'filterParams', 'config']);
      }
      case 'object': {
        return cleanTree(item, ['id', 'description']);
      }
      case 'service': {
        return cleanTree(item, ['id', 'description']);
      }
      case 'funcComp': {
        return cleanTree(item, [
          'id',
          'description',
          'filterParams',
          'config',
          'objectArray',
        ]);
      }
      default:
        break;
    }
  });
};
