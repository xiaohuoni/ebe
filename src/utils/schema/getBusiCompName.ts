import changeCase from 'change-case';
import pinyin from 'pinyin';

const busiCompMap: any = {};
const busiCompNameMap: any = {};
export const getBusiCompName = (busiCompMapping: any, node: any) => {
  const otherType = busiCompMapping?.[node?.props?.busiCompId] ?? '';
  try {
    if (busiCompNameMap[otherType]) {
      return busiCompNameMap[otherType];
    }
    const pageName = node?.pageName ?? node?.label;
    const name = pinyin(`${pageName}`, {
      style: 'normal',
    })
      .map((i) => [changeCase.pascalCase(i[0])])
      .join('')
      .replaceAll('/', '')
      .replaceAll('(', '')
      .replaceAll(')', '');
    if (busiCompMap[name]) {
      const _name =
        busiCompMap[name] === otherType ? name : `${name}${otherType}`;
      busiCompNameMap[otherType] = _name;
      return _name;
    }
    busiCompNameMap[otherType] = name;
    busiCompMap[name] = otherType;
    return name;
  } catch (error) {
    return `BusiComp${otherType}`;
  }
};
