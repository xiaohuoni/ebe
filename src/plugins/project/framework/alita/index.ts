import template from './template';
import packageJSON from './plugins/packageJSON';
import mobileLayout from './plugins/mobileLayout';
import commonDeps from './plugins/commonDeps';
import config from './plugins/config';
import hackEngineApis from './hooks/hackEngineApis';

export default {
  template,
  plugins: {
    packageJSON,
    mobileLayout,
    commonDeps,
    config,
  },
  hooks: {
    hackEngineApis,
  },
};
