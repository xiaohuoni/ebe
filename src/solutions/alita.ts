import {
  createProjectBuilder,
  FileType,
  IProjectBuilder,
  IProjectBuilderOptions,
  LXProjectOptions,
} from '../core';
import esModule from '../core/plugins/common/esmodule';
import { babel, fixUndefinedVar, prettier } from '../core/postprocessor';
import { SchemaParser } from '../parser/SchemaParser';
import alita from '../plugins/project/framework/alita';
import reactContainer from '../plugins/react/container';
import containerHook from '../plugins/react/containerHook';
// import containerLifeCycle from '../plugins/react/containerLifeCycle';
import customActionMap from '../plugins/react/customActionMap';
import reactJsx from '../plugins/react/jsx';
import reactCommonDeps from '../plugins/react/reactCommonDeps';

export type AlitaProjectBuilderOptions = IProjectBuilderOptions;

// options.appConfig.frontendHookList
// options.appConfig.frontendHookList[0]
// options.appConfig.frontendHookList[0].hookCode = "fetchSuccess"
// options.appConfig.frontendHookList[0].hookCompiledContent = "(function(data,_ref,_ref2,_ref3){var message=_ref.message,notification=_ref.notification,modal=_ref.modal;var url=_ref2.url,params=_ref2.params;var CryptoJS=_ref3.CryptoJS;console.log(\"\\u8BF7\\u6C42\\u6210\\u529F\");return data});"
// options.appConfig.frontendHookList[0].hookSourceContent ="/**\n * @param {object} data 响应业务数据\n * @param {object} message 全局消息提示\n * @param {object} notification 通知提示框\n * @param {object} modal 对话框\n * @param {string} url 请求地址\n * @param {object} params 请求参数\n * @param {object} CryptoJS 加密工具\n * @return {object} 返回请求成文功报\n*/\n(data, { message, notification, modal }, { url, params }, { CryptoJS }) => {\n    console.log('请求成功')\n    return data\n}\n"
// options.appConfig.frontendHookList[0].hookType = 'req'
const getFrontendHookList = (frontendHookList: any[]) => {
  const returnObject: any = {};
  frontendHookList.forEach((item) => {
    if (item?.hookCode && item.hookSourceContent) {
      // TODO: 根据不同的钩子有不同的模版，可以增加入参类型
      if (item.hookSourceContent.includes('\n(')) {
        returnObject[item?.hookCode] = item.hookSourceContent.replace(
          '\n(',
          `\n const ${item?.hookCode} = (`,
        );
      }
    }
  });

  return returnObject;
};

export default function createIceJsProjectBuilder(
  options?: AlitaProjectBuilderOptions,
): IProjectBuilder {
  return createProjectBuilder({
    inStrictMode: options?.inStrictMode,
    extraContextData: {
      ...options,
      options: {
        ...options?.options,
        frontendHookMap: getFrontendHookList(
          options?.options?.appConfig?.frontendHookList ?? [],
        ),
      } as LXProjectOptions,
    },
    template: alita.template,
    schemaParser: new SchemaParser(),
    plugins: {
      pages: [
        containerHook(),
        customActionMap(),
        reactCommonDeps(),
        alita.plugins.mobileLayout(),
        alita.plugins.styleInject(),
        reactContainer(),
        // TODO: 临时注释，后续稳定后移除
        // containerLifeCycle(),
        reactJsx({
          nodePlugins: [alita.hooks.hackNodeAttr],
          nodeTypeMapping: {
            Page: 'div',
            MobileModal: 'div',
            MobilePopover: 'div',
            Modal: 'div',
            Drawer: 'div',
            BusiComp: 'div',
          },
        }),
        // 最前面的代码，其实是最后生成的
        esModule({
          fileType: FileType.TSX,
        }),
        alita.plugins.dataSource(),
        alita.plugins.globalData(),
      ],
      pageview: [
        esModule({
          fileType: FileType.TSX,
        }),
        alita.plugins.pageview(),
      ],
      app: [alita.plugins.app()],
      appConfig: [alita.plugins.config()],
      packageJSON: [
        alita.plugins.packageJSON({
          useEbeSetup: options?.useEbeSetup !== false,
          dependenciesPc: {
            '@lingxiteam/charts-common': '^3.7.1-alpha.10',
            '@lingxiteam/lcdp-ueditor-react': '^1.0.0-alpha.15',
            '@lingxiteam/load-framework': '0.2.1',
            '@lingxiteam/richtext': '0.5.1',
            echarts: '^5.1.2',
            'lcdp-message': '0.1.9',
            'qr-code-with-logo': '1.1.0',
            'rc-tween-one': '3.0.6',
            'react-player': '2.11.0',
            'react-quill': '^2.0.0',
            'react-resizable': '^3.0.4',
            'react-soundplayer': '^1.0.5',
            'size-sensor': ' ^1.0.1',
            antd: '4.24.13',
            lodash: '^4.17.21',
            'copy-to-clipboard': '^3.3.1',
            '@lingxiteam/icons': '0.0.3',
            '@lingxiteam/types': '3.3.1-alpha.14',
            merge: '^2.1.1',
            uuid: '^9.0.1',
            dayjs: '1.11.3',
            '@ant-design/icons': '^4.2.2',
            'react-loadable': '^5.5.0',
            'react-pdf': '4.2.0',
            moment: '^2.30.1',
            'es6-proxy-polyfill': '^2.1.1',
          },
          dependencies: {
            '@lingxiteam/cli': '^0.4.6',
            'antd-mobile-5': '5.26.0',
            'react-loadable': '^5.5.0',
            merge: '^2.1.1',
            uuid: '^9.0.1',
            dayjs: '1.11.3',
            'browserify-zlib': '^0.2.0',
            'es6-proxy-polyfill': '^2.1.1',
            '@lingxiteam/antd-mobile-plus': '0.5.0',
            '@lingxiteam/antd-mobile-v2-theme': '0.0.1',
            '@lingxiteam/charts': '0.0.8',
            '@lingxiteam/charts-common': '3.7.1',
            '@lingxiteam/dform': '3.8.1-alpha.4',
            '@lingxiteam/load-framework': '0.2.1',
            '@paciolan/remote-component': '2.12.1',
            'ag-grid-community': '30.0.2',
            'ag-grid-react': '30.0.2',
            echarts: '^5.1.2',
            'lcdp-message': '0.1.9',
            'rc-field-form': '^1.27.0',
            'rc-tween-one': '3.0.6',
            'react-player': '2.9.0',
            'react-soundplayer': '^1.0.5',
            'rmc-tabs': '~1.2.0',
            '@alitajs/f2': '^0.1.1',
            '@lingxiteam/icons': '0.0.3',
            'rc-animate': '^2.4.4',
            ahooks: '^3.7.7',
            '@js-preview/excel': '^1.4.5',
            'docx-preview': '0.1.20',
            'react-pdf': '4.2.0',
            'copy-to-clipboard': '^3.3.1',
            'pdfjs-dist': '2.1.266',
          },
          devDependencies: {
            'copy-webpack-plugin': '5.1.1',
            '@types/qs': '^6.9.6',
            '@types/lodash': '^4.14.162',
          },
        }),
      ],
      models: [alita.plugins.models()],
    },
    postProcessors: [
      babel(),
      fixUndefinedVar({
        // 只需要处理pages/components目录
        includes: [/^src\/(pages|components)/],
      }),
      prettier(),
    ],
    // postProcessors: [],
    printUtil: options?.printUtil,
  });
}
