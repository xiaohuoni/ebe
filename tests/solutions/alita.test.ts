import alita from '../../src/solutions/alita';
import schema from './n.json';

describe('solutions/alita', () => {
  test('compilation completed', async () => {
    const result = alita({
      options: schema.options as any,
      useEbeSetup: false,
    });
    expect(result).toMatchSnapshot();
  });
});
