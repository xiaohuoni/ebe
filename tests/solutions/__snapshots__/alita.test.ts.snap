// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`solutions/alita compilation completed 1`] = `
ProjectBuilder {
  "extraContextData": Object {
    "options": Object {
      "appConfig": Object {
        "appCode": "APP_868681578956083200",
        "appId": "868681578956083200",
        "appMenuJson": null,
        "appName": "集客成员营销任务管理",
        "applicationIcon": null,
        "frameExtension": Object {},
        "frontendHookList": Array [
          Object {
            "hookCode": "appDidInit",
            "hookCompiledContent": "(function(appInfo){window.scqx_fetch_num=0;document.title=\\"\\\\u96C6\\\\u5BA2\\\\u5168\\\\u606F\\\\u4E13\\\\u533A\\";window.parent.document.title=document.title;var script=document.createElement(\\"script\\");script.src=\\"https://api.map.baidu.com/library/GeoUtils/1.2/docs/symbols/src/BMapLib_GeoUtils.js\\";document.body.appendChild(script)});",
            "hookSourceContent": "/**
 * @param {object} appInfo 应用信息
*/
(appInfo) => {
  window.scqx_fetch_num = 0
  document.title = '集客全息专区';
  window.parent.document.title = document.title;
  const script = document.createElement('script');
  script.src = 'https://api.map.baidu.com/library/GeoUtils/1.2/docs/symbols/src/BMapLib_GeoUtils.js';
  document.body.appendChild(script);
}
",
            "hookType": "app",
          },
          Object {
            "hookCode": "fetchResponse",
            "hookCompiledContent": "(function(response,httpCode,success,fail,_ref,_ref2){var url=_ref.url,params=_ref.params;var CryptoJS=_ref2.CryptoJS;window.scqx_fetch_num=window.scqx_fetch_num-1});",
            "hookSourceContent": "/**
 * @param {object} response 响应报文
 * @param {number} httpCode http响应编码
 * @param {function} success 触发成功，传入报文业务数据
 * @param {function} fail 触发失败，传入错误信息
 * @param {string} url 请求地址
 * @param {object} params 请求参数
 * @param {object} CryptoJS 加密工具
*/
(response, httpCode, success, fail, { url, params }, { CryptoJS }) => {
    // if(window.location.href.includes('zhongzhuanyemian4662')){
     window.scqx_fetch_num=window.scqx_fetch_num-1
  // }
}
",
            "hookType": "req",
          },
          Object {
            "hookCode": "fetchSendBefore",
            "hookCompiledContent": "(function(url,method,params,_ref){var CryptoJS=_ref.CryptoJS;window.scqx_fetch_num=window.scqx_fetch_num+1});",
            "hookSourceContent": "/**
 * @param {string} url 请求地址
 * @param {string} method 请求方法
 * @param {object} params 请求参数
 * @param {object} CryptoJS 加密工具
 * @return {object} {
    header:'附带请求头数据',
    params:'参数',
    url: '改写请求地址',
    fetchOption: 'fetch支持的配置',
  } 返回对象，请求层将对请求配置二次处理
  * @return {boolean} false 中止该请求
  * @return {promise} 不执行网络请求，但会根据Promise执行 ‘请求成功’和‘请求失败’
*/
(url, method, params, { CryptoJS }) => {
  // if(window.location.href.includes('zhongzhuanyemian4662')){
    window.scqx_fetch_num=window.scqx_fetch_num+1
  // }
}
",
            "hookType": "req",
          },
        ],
        "globalDataSource": null,
        "i18nInfo": Object {
          "enabled": null,
          "locales": null,
        },
      },
      "appId": "868681578956083200",
      "baseUrl": "http://10.10.193.65:9012/",
      "busiCompMapping": Object {},
      "compAssetList": Array [],
      "frontendHookMap": Object {
        "appDidInit": "/**
 * @param {object} appInfo 应用信息
*/
 const appDidInit = (appInfo) => {
  window.scqx_fetch_num = 0
  document.title = '集客全息专区';
  window.parent.document.title = document.title;
  const script = document.createElement('script');
  script.src = 'https://api.map.baidu.com/library/GeoUtils/1.2/docs/symbols/src/BMapLib_GeoUtils.js';
  document.body.appendChild(script);
}
",
        "fetchResponse": "/**
 * @param {object} response 响应报文
 * @param {number} httpCode http响应编码
 * @param {function} success 触发成功，传入报文业务数据
 * @param {function} fail 触发失败，传入错误信息
 * @param {string} url 请求地址
 * @param {object} params 请求参数
 * @param {object} CryptoJS 加密工具
*/
 const fetchResponse = (response, httpCode, success, fail, { url, params }, { CryptoJS }) => {
    // if(window.location.href.includes('zhongzhuanyemian4662')){
     window.scqx_fetch_num=window.scqx_fetch_num-1
  // }
}
",
        "fetchSendBefore": "/**
 * @param {string} url 请求地址
 * @param {string} method 请求方法
 * @param {object} params 请求参数
 * @param {object} CryptoJS 加密工具
 * @return {object} {
    header:'附带请求头数据',
    params:'参数',
    url: '改写请求地址',
    fetchOption: 'fetch支持的配置',
  } 返回对象，请求层将对请求配置二次处理
  * @return {boolean} false 中止该请求
  * @return {promise} 不执行网络请求，但会根据Promise执行 ‘请求成功’和‘请求失败’
*/
 const fetchSendBefore = (url, method, params, { CryptoJS }) => {
  // if(window.location.href.includes('zhongzhuanyemian4662')){
    window.scqx_fetch_num=window.scqx_fetch_num+1
  // }
}
",
      },
      "pageIdMapping": Object {
        "/TTTC1858": "899834190456836096",
        "/TTTTTTT0978": "897865290545934336",
        "/TTTTTTT0978copy": "955659038239764480",
        "/TracyTest1247": "869780141765591040",
        "/applicationFormSetList?appId=868681578956083200&catalogItemId=882206099849510912&pageId=882206104794595328&type=page": "882206104794595328",
        "/biaogejiazaitupianceshi0015123": "956383215578001408",
        "/biaozhunchanpinyingxiao2435": "1019132055831023616",
        "/ceshibiaogejiazaitupian5316": "956029665732726784",
        "/ceshiyemian012332": "899118936410267648",
        "/changjingxuanzeyemian2411": "876720646100758528",
        "/daijiandangqiye2910": "1045238349683339264",
        "/daipaifajituanfenbuxiangqingtanchuang3926": "869758859921620992",
        "/daoru6125": "871590497857712128",
        "/diaoyongfuchuangkoufangfa7521": "1067800877664919552",
        "/gequxianjituanduiying": "927743380699754496",
        "/gongdanchuli9650": "971613994714750976",
        "/gongdanfaqiguanlijiemian2425copycopy": "889083791716089856",
        "/gongdanfaqiguanlijiemian2425copycopycopy": "945580960623509504",
        "/gongdangmingxitongjibaobiao": "909615118796615680",
        "/gongdanpaifa6251": "871281670436302848",
        "/gongdanpaifa62511copy": "907922822368804864",
        "/gongdanpaifa6251copy": "1002115088770547712",
        "/gongdanpicipaifatongjibaobiao145": "953538989051117568",
        "/gongdanpicipaifatongjibaobiao1454": "912596729668415488",
        "/gongdanpicipaifatongjibaobiao1454copy": "946304764975820800",
        "/gongdanpicipaifatongjibaobiao1454copycopy": "951013791857922048",
        "/gongdanpicipaifatongjibaobiao145copy": "956454499648569344",
        "/gongdanpicipaifatongjibaobiao145copycopy": "956879786506088448",
        "/gongdanrenyuanpeizhi5800": "891908709324177408",
        "/gongdanrenyuanpeizhi5800-new": "903211721105510400",
        "/gongdanrenyuanpeizhi5800copy": "892221456175607808",
        "/gongdanrenyuanpeizhi5800copycopycopy": "898472894087344128",
        "/gongdanzhongzhitanchuang7731": "1008926366107435008",
        "/hangyezongjian6654": "899133742030757888",
        "/haomaqingdan0168": "981131908525371392",
        "/huaxiangmingxi8861": "942982691350474752",
        "/imageViewer": "955842601212207104",
        "/imageViewerWindow": "956490978775220224",
        "/imageViewercopy": "956487143990181888",
        "/jikechengyuangongjian": "903831414833057792",
        "/jikechengyuangongjiancopy": "927873289162076160",
        "/jikeliuchengfaqiqishihuanjie5389": "874125593343266816",
        "/jikequanxipaoxiaomingxibaobiao3323": "897003315829030912",
        "/jikequanxitongtanmingxibaobiao0957": "896941245100081152",
        "/jikequanxizhuanhualvbaobiao5891": "896941025859616768",
        "/jikezhongkegongdanpaifa6251copy": "898455841213296640",
        "/jikezongjianshenpi-modihou4964": "881738185543417856",
        "/jituanbiaoqiankoujing3998": "963066086472966144",
        "/jituanbiaoqiankoujing3998copy": "964111772751933440",
        "/jituanbiaoqiankoujing3998copycopy": "969141912905732096",
        "/jituanchengyuanqingdan": "916882232428281856",
        "/jituandaorucuowutishi1190": "900029343968677888",
        "/jituankehuxuanzetanchuangyemian6218": "870565605452427264",
        "/jituanxinxi2650": "943040967534538752",
        "/jituanxinxi4636": "942243904962838528",
        "/jituanzhongkegongdanguiji6251": "898456240108384256",
        "/kehujinglimodichulihuanjie6349": "874130140828635136",
        "/kehujinglimodichulihuanjie6349copy": "881790219080597504",
        "/liuchengdiaoduyichanggongdanchaxun9364": "942251215550824448",
        "/luyinzhijian4049": "963686167003422720",
        "/msceshibiaoge4226": "1002111390342938624",
        "/paifachuliren8124": "902388256959332352",
        "/paoxiao5207": "978572536241909760",
        "/paoxiao5207copy": "990550162314813440",
        "/piliangdaoruhaoma1065": "981021638165823488",
        "/renwuzhixing4449": "954996620153348096",
        "/shaixuanjituan8136copy": "983192562538672128",
        "/shijianceshi1230058": "914766039526518784",
        "/shuangdigongdanguijixiangxi6251": "901047083795439616",
        "/testzzh12425copycopycopy": "902484901092884480",
        "/tongtan8515": "973509311629885440",
        "/tongtan8743copy": "873146627329892352",
        "/tongtan8743copycopy": "873525059763052544",
        "/tongtan8743copycopycopy": "927936604223582208",
        "/tongtanjikezongjianshenpifenzhipanduantiaojian7895": "881805735010664448",
        "/tongtanjikezongjianshenpifenzhipanduantiaojian7895copy": "901099424242425856",
        "/ttplan": "976028589426262016",
        "/tupianzhanshi5631": "956870760006053888",
        "/woderenwu2258copy": "898435362863755264",
        "/xinjikeshuangdichenggonggongjiangongdanfaqi9076": "954928233963356160",
        "/xuanzehaoma8158": "872310912473243648",
        "/xuanzejikezongjian2731": "979301546524704768",
        "/xuanzejikezongjian2731copy": "980647618656370688",
        "/yewuxiangqing2512": "909617376865140736",
        "/yingxiaotongtanjindujituanweidu4023": "940417506565959680",
        "/zhixiaoduizuzhixuanze5800": "912268597019127808",
        "/zhuanhualvbaobiao5891": "911867890456674304",
        "/zuozhanzhihuishenpifenzhipanduantiaojian4403": "901043509225021440",
      },
      "platform": "pc",
    },
    "useEbeSetup": false,
  },
  "hooks": Hookable {
    "_after": Array [
      [Function],
    ],
    "_before": undefined,
    "_deprecatedHooks": Object {},
    "_deprecatedMessages": undefined,
    "_hooks": Object {},
    "callHook": [Function],
    "callHookWith": [Function],
    "hook": [Function],
  },
  "inStrictMode": false,
  "plugins": Object {
    "app": Array [
      [Function],
    ],
    "appConfig": Array [
      [Function],
    ],
    "models": Array [
      [Function],
    ],
    "packageJSON": Array [
      [Function],
    ],
    "pages": Array [
      [Function],
      [Function],
      [Function],
      [Function],
      [Function],
      [Function],
      [Function],
      [Function],
      [Function],
      [Function],
    ],
    "pageview": Array [
      [Function],
      [Function],
    ],
    "themeCss": Array [
      [Function],
    ],
  },
  "postProcessors": Array [
    [Function],
    [Function],
    [Function],
  ],
  "printUtil": Object {
    "prettier": [Function],
  },
  "projectPostProcessors": Array [],
  "projectPreProcessors": Array [],
  "schemaParser": SchemaParser {},
  "template": Object {
    "generateTemplate": [Function],
    "slots": Object {
      "app": Object {
        "fileName": "app",
        "path": Array [
          "src",
        ],
      },
      "appConfig": Object {
        "fileName": "config",
        "path": Array [
          "config",
        ],
      },
      "components": Object {
        "path": Array [
          "src",
          "components",
          "businessComponent",
        ],
      },
      "drawerPages": Object {
        "path": Array [
          "src",
          "pages",
          "drawer",
        ],
      },
      "modalPages": Object {
        "path": Array [
          "src",
          "pages",
          "modal",
        ],
      },
      "packageJSON": Object {
        "fileName": "package",
        "path": Array [],
      },
      "pages": Object {
        "path": Array [
          "src",
          "pages",
        ],
      },
      "pageview": Object {
        "path": Array [
          "src",
          "components",
          "common",
          "Pageview",
        ],
      },
      "popoverPages": Object {
        "path": Array [
          "src",
          "pages",
          "popover",
        ],
      },
      "themeCss": Object {
        "fileName": "overrides",
        "path": Array [
          "src",
        ],
      },
    },
  },
}
`;
